### PR TITLE
fix: specify supported versions of php 8

### LINKED ISSUE #14174: [Improvement]: Clarify php 8.2 support for pimcore 10.x
### Improvement description

Currently the documentation (https://github.com/pimcore/pimcore/blob/10.5/doc/Development_Documentation/23_Installation_and_Upgrade/01_System_Requirements.md#php--80) and the `composer.json` of pimcore 10.x (https://github.com/pimcore/pimcore/blob/10.5/composer.json#L31) suggest that it's compatible with php 8.2.

But this open PR (https://github.com/pimcore/pimcore/pull/14080) targeted to pimcore 11 and this comment (https://github.com/pimcore/pimcore/pull/13861#issuecomment-1361307701) suggest otherwise.
* If pimcore 10.x is already compatible with php 8.2: feel free to close this issue
* Otherwise I suggest updating the docs and `composer.json` accordingly (I'm happy to provide a PR for this)

### PR DESCRIPTION
**Pull request by juckerf**
closes #14174

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
     "docs": "https://pimcore.com/docs/latest/"
   },
   "require": {
-    "php": "^8.0",
+    "php": ">=8.0 <8.2",
```
Comment (jdreesen) [inline composer.json]: I propose:

```suggestion
    "php": "~8.0.0 || ~8.1.0",
```

**Comment by jdreesen** [comment]
I don't think this breaks anything, as composer just won't install any new Pimcore 10 versions when you use PHP 8.2.

**Comment by juckerf** [comment]
> I don't think this breaks anything, as composer just won't install any new Pimcore 10 versions when you use PHP 8.2.

I guess you're right. Haven't thought that through.

**Comment by brusch** [comment]
@jdreesen yep and it won't run anyway on PHP 8.2 - so there shouldn't be any existing project using it 😉

**Comment by brusch** [comment]
Thank you very much!

**Comment by juckerf** [comment]
thanks for the merge!

**Comment by ZaneCEO** [comment]
hi guys! I'm a bit confused here. This change looks sound and fixes my issue. But I cannot find it even if the PR has been merged. It looks like it reverted here https://github.com/pimcore/pimcore/commit/68f60a35d0643b98fc6fa0301f501707104d6566 ....?

**Comment by brusch** [comment]
@ZaneCEO don't know what you mean, it's still correct on the `10.5` branch, see: 
https://github.com/pimcore/pimcore/blob/10.5/composer.json#L31

**Comment by ZaneCEO** [comment]
hi @brusch ! Sorry, my bad. I misunderstood the issue (twice)

**Pull request merged**